gobblin-data-management/src/main/java/org/apache/gobblin/data/management/copy/hive/UnpartitionedTableFileSet.java

```diff
@@ -64,11 +65,21 @@ public UnpartitionedTableFileSet(String name, HiveDataset dataset, HiveCopyEntit
 
     Optional<Table> existingTargetTable = this.helper.getExistingTargetTable();
     if (existingTargetTable.isPresent()) {
-      if (!this.helper.getTargetTable().getDataLocation().equals(existingTargetTable.get().getDataLocation())) {
+      boolean path_mismatch = false;
+      try {
+        if (!this.helper.getTargetFs().resolvePath(this.helper.getTargetTable().getDataLocation())
+            .equals(this.helper.getTargetFs().resolvePath(existingTargetTable.get().getDataLocation()))) {
+          path_mismatch = true;
+        }
+      } catch (FileNotFoundException e) {
+        // If desired path does not exist, then user is defining a different snapshot path so check policy
+        path_mismatch = true;
+      }
+      if (path_mismatch) {
```

Comment:
I was actually contemplating similar (boolean flag) above, when I saw two code paths to the same exception thrown.  since java has no widespread, canonical lib for converting exception control flow into values (like scala's https://www.scala-lang.org/api/2.12.4/scala/util/control/Exception$.html ) there's no simple way to phrase this.

as this already recurs twice, I'd seek a utility abstraction.  maybe just a static method taking a `FileSystem` and two 'locations' (what we call `.getDataLocation()` on).  it would be an equality predicate (capturing any exception within and converting to `false`).

Reply:
Ah good suggestion, will implement it this way